index.js

```diff
@@ -17,11 +17,19 @@ const buildPathObjects = paths => _.map(paths, (pathDef, path) => ({
   pathDef,
 }));
 
+const stripBasePath = (url) => {
+    const basePath = options.schema.basePath;
+
+    if (basePath && (url.indexOf(basePath) == 0)) {
+        return url.slice(basePath.length);
+    } else {
+        return url;
+    }
+};
+
 const matchUrlWithSchema = (reqUrl) => {
-  let url = parseUrl(reqUrl).pathname;
-  if (options.schema.basePath) {
-    url = url.replace(options.schema.basePath, '');
```

Comment:
The original code will replace occurrences of the basepath that occur in the middle of the input URL. (As long as they are the first such instance.)

```
> const basePath = '/b';
undefined
> '/a/b/c/d'.replace(basePath, '')
'/a/c/d'
```